### ISSUE TITLE
Lost log messages in docker logs

### ISSUE THREAD

**Issue by scolby33** [phusion/baseimage-docker#436]
I have been having trouble with missing log messages and I think I've tracked down the issue.

I have a service that logs to stdout and an appendant log service like so:

```shell
#!/bin/sh
exec /usr/bin/logger -t servicename
```

In some cases (I think because it's a race condition), the startup messages from this service were not propagated to docker-logs. At first I tried adding `sv start syslog-forwarder || exit 1` to the beginning of the log service run file to make sure the forwarder was running before any log messages were processed. Unfortunately, the inconsistent behavior continued.

I thought perhaps the start order of the forwarder and syslog-ng itself mattered, so I prepended the following line to `/etc/service/syslog-forwarder/run`:

```shell
sv start syslog-ng || exit 1
```

With both of these changes in place, the startup messages from my service are now consistently **not** lost. I'm not sure exactly why syslog-ng needs to be running before the forwarder...the opposite would be my instinct, but I cannot argue with the positive results. In fact, with this change made, the syslog-ng startup message always is captured by docker-logs as well.

Perhaps this change should be added to baseimage-docker?

**Comment by Theaxiom** [comment]
Certainly, would you mind submitting a PR? Thanks!

**Comment by scolby33** [comment]
I would love to! Before doing that though, I would just like to make sure the solution makes sense. As I said, it doesn't make complete sense to me that starting the syslog daemon before the forwarder would solve the problem.

A sanity check of the logic would be much appreciated--I don't want to introduce another race condition :)

**Comment by scolby33** [comment]
Something weird is still going on. The first time the container is run with `docker run` some log messages are going missing. Subsequent runs with `docker stop ...; sleep 2; docker start ...` don't show the issue. Interestingly, `docker stop ...; docker start ...` does show the issue. I'm going to try and create a minimal working example so others can test.

**Comment by scolby33** [comment]
Okay, here's a MWE exhibiting the problem.

Dockerfile:
```dockerfile
FROM phusion/baseimage:0.9.22

# install minidaemon
COPY minidaemon.sh /usr/local/bin/minidaemon.sh
RUN chmod +x /usr/local/bin/minidaemon.sh

# add minidaemon as a service for runit
RUN mkdir -p /etc/service/minidaemon/log
COPY minidaemon-run.sh /etc/service/minidaemon/run
COPY minidaemon-log.sh /etc/service/minidaemon/log/run
RUN chmod +x /etc/service/minidaemon/run /etc/service/minidaemon/log/run

CMD ["/sbin/my_init"]
```

minidaemon.sh:
```bash
#!/bin/bash

trap "echo SIGHUP" HUP
trap "echo Shutting down; exit" TERM

echo "Startup message"
echo "Performing initial work"
echo "Other things"

while :
do
        sleep 1
done
```

minidaemon-run.sh:
```shell
#!/bin/sh
exec 2>&1
exec /usr/local/bin/minidaemon.sh
```

minidaemon-log.sh:
```shell
#!/bin/sh
exec /usr/bin/logger -t minidaemon
```

And here's how running it goes:
```console
# docker build -t logissue .
# docker run -it -d --name logissue logissue
# docker logs logissue
*** Running /etc/my_init.d/00_regen_ssh_host_keys.sh...
*** Running /etc/rc.local...
*** Booting runit daemon...
*** Runit started as PID 9
# docker exec -it logissue sv hup minidaemon
# docker logs logissue
*** Running /etc/my_init.d/00_regen_ssh_host_keys.sh...
*** Running /etc/rc.local...
*** Booting runit daemon...
*** Runit started as PID 9
```
Notice how neither the startup messages nor the SIGHUP alert show up in the log.

The minimum needed to get the forwarder working appears to be:
```console
# docker exec -it logissue sv restart syslog-forwarder
# docker exec -it logissue sv restart minidaemon/log
# docker exec -if logissue sv hup minidaemon
# docker logs logissue
*** Running /etc/my_init.d/00_regen_ssh_host_keys.sh...
*** Running /etc/rc.local...
*** Booting runit daemon...
*** Runit started as PID 9
Sep 21 06:32:06 b75c817c0b1c minidaemon: SIGHUP
```
Restarting just `syslog-forwarder` or just `minidaemon-log` doesn't do the trick; however, the order of restarting them doesn't seem to matter.

After these restarts, you can also see the shutdown and startup messages:
```console
# docker exec -it logissue sv restart minidaemon
# docker logs logissue
*snip*
Sep 21 06:45:32 8686b048ca7f minidaemon: Shutting down
Sep 21 06:45:32 8686b048ca7f minidaemon: Startup message
Sep 21 06:45:32 8686b048ca7f minidaemon: Performing initial work
Sep 21 06:45:32 8686b048ca7f minidaemon: Other things
```

You also get syslog-ng's own messages, too:
```console
# docker exec -it logissue sv restart syslog-ng
# docker logs logissue
*snip*
Sep 21 06:47:58 8686b048ca7f syslog-ng[18]: syslog-ng shutting down; version='3.5.6'
Sep 21 06:47:58 8686b048ca7f syslog-ng[235]: syslog-ng starting up; version='3.5.6'
```

I haven't let it run long enough to see what happens to cron's log messages.

At this point I'm not exactly sure what the proper solution is, but there's definitely something a bit broken here when using syslog-ng, syslog-forwarder, and the `logger` command to log to syslog.

**Comment by Theaxiom** [comment]
This issue has been resolved by #425
